lms/djangoapps/instructor/views/api.py

```diff
@@ -1267,7 +1267,11 @@ def generate_registration_codes(request, course_id):
     csv_file = StringIO.StringIO()
     csv_writer = csv.writer(csv_file)
     for registration_code in registration_codes:
-        csv_writer.writerow([registration_code.code])
+        full_redeem_code_url = '{base_url}{redeem_code_url}'.format(
```

Comment:
Not sure if this will always work. I see on line 1222 site_name can default to localhost, which means you may not get a full qualified URL. 

Using the request, you could use request.build_absolute_uri() and pass the reversed URL for the code redemption
https://docs.djangoproject.com/en/dev/ref/request-response/#django.http.HttpRequest.build_absolute_uri

That may be more stable.